src/solver.rs

```diff
@@ -215,7 +218,17 @@ pub enum Dependencies<P: Package, VS: VersionSet> {
     /// Package dependencies are unavailable.
     Unknown,
     /// Container for all available package versions.
-    Known(DependencyConstraints<P, VS>),
+    Known(DependencyConstraints<P, Requirement<VS>>),
+}
+
+/// An enum that defines the type of dependency.
+#[derive(Clone, Debug)]
+pub enum Requirement<VS: VersionSet> {
```

Comment:
this will need to `#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]`, and it's going to need to be even a little more complicated than that. because it needs to deserialize the existing test files and benchmarks, which do not have a marker for which kind it is.

Reply:
Or we could decide to break compatibility of serialized data? or start defining a serialized format more in line with the internal representation of incompatiblities (list of pairs of pkg, and +/- version set) which is not going to change soon as this would not be a "pubgrub" algorithm anymore? Then will have manually written serialize and deserializers.